examples/data_types_and_io/data_types_and_io/structured_dataset.py

```diff
@@ -153,3 +155,114 @@ def numpy_wf() -> Annotated[StructuredDataset, None, PARQUET]:
     print(f"A simple Pandas dataframe workflow: {sd.open(pd.DataFrame).all()}")
     print(f"Using CSV as the serializer: {pandas_to_csv_wf().open(pd.DataFrame).all()}")
     print(f"NumPy encoder and decoder: {numpy_wf().open(np.ndarray).all()}")
+
+data = [
+    {
+        "company": "XYZ pvt ltd",
+        "location": "London",
+        "info": {"president": "Rakesh Kapoor", "contacts": {"email": "contact@xyz.com", "tel": "9876543210"}},
+    },
+    {
+        "company": "ABC pvt ltd",
+        "location": "USA",
+        "info": {"president": "Kapoor Rakesh", "contacts": {"email": "contact@abc.com", "tel": "0123456789"}},
+    },
+]
+
+
+@dataclass
+class ContactsField:
+    email: str
+    tel: str
+
+
+@dataclass
+class InfoField:
+    president: str
+    contacts: ContactsField
+
+
+@dataclass
+class CompanyField:
+    location: str
+    info: InfoField
+    company: str
+
+
+MyArgDataset = Annotated[StructuredDataset, kwtypes(company=str)]
+MyTopDataClassDataset = Annotated[StructuredDataset, CompanyField]
+MyTopDictDataset = Annotated[StructuredDataset, {"company": str, "location": str}]
+
+MyDictDataset = Annotated[StructuredDataset, kwtypes(info={"contacts": {"tel": str}})]
+MyDictListDataset = Annotated[StructuredDataset, kwtypes(info={"contacts": {"tel": str, "email": str}})]
+MySecondDataClassDataset = Annotated[StructuredDataset, kwtypes(info=InfoField)]
+MyNestedDataClassDataset = Annotated[StructuredDataset, kwtypes(info=kwtypes(contacts=ContactsField))]
+
+
+@task()
```

Comment:
Could we also specify an ImageSpec in @task? This ImageSpec should install `tabulate`

Reply:
done!